Sources/ElasticSwift/Requests/DeleteRequest.swift

```diff
@@ -49,113 +53,57 @@ public class DeleteRequestBuilder: RequestBuilder {
         return self
     }
     
-    public func set(completionHandler: @escaping (DeleteResponse?, Error?) -> Void) -> Self {
+    public func set(completionHandler: @escaping (DeleteResponse?, Error?) -> ()) -> Self {
         self.completionHandler = completionHandler
         return self
     }
     
-    public func build() -> Request {
+    public func make() throws -> DeleteRequest {
         return DeleteRequest(withBuilder: self)
     }
-    
+
 }
 
 public class DeleteRequest: Request {
+   
+    public typealias ResponseType = DeleteResponse
     
-    let client: ESClient
+    public var completionHandler: ((DeleteResponse?, Error?) -> ())?
+    
+    public let client: ESClient
     let index: String
-    let type: String
+    let type: String?
     let id: String
     var version: Int?
-    var completionHandler: ((DeleteResponse?, Error?) -> Void)
     
     init(withBuilder builder: DeleteRequestBuilder) {
         self.client = builder.client
-        self.index = builder.index!
-        self.type = builder.type!
-        self.id = builder.id!
+        self.index = builder.index
+        self.id = builder.id
+        
+        self.type = builder.type
         self.version = builder.version
-        self.completionHandler = builder.completionHandler!
+        self.completionHandler = builder.completionHandler
     }
     
-    public var method: HTTPMethod {
-        get {
-            return .DELETE
-        }
-    }
+    public var method: HTTPMethod = .DELETE
     
     public var endPoint: String {
         get {
-            return makeEndPoint()
-        }
-    }
-    
-    public var body: Data {
-        get {
-            return Data()
-        }
-    }
-    
-    public func execute() {
-        self.client.execute(request: self, completionHandler: responseHandler)
-    }
-    
-    func makeEndPoint() -> String {
-        return self.index + "/" + self.type + "/" + self.id
-    }
-    
-    func responseHandler(_ response: ESResponse) -> Void {
```

Comment:
This is managed centrally in the protocol extension as base implementation for all requests